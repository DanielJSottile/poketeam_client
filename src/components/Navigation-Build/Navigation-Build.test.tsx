import React, { Suspense } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
//import { render } from '@testing-library/react';
import NavigationBuild from './Navigation-Build';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<NavigationBuild/>', (): void => {
  it('renders without crashing', (): void => {
    const div: HTMLDivElement = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Suspense fallback={<div>test</div>}>
          <NavigationBuild />
        </Suspense>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('snapshot', (): void => {
    const wrapper = shallow(<NavigationBuild />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
