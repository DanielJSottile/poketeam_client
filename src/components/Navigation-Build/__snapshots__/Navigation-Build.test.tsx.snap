// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBuild/> snapshot 1`] = `
<div>
  <nav
    role="navigation"
  >
    <div
      className="user-welcome"
    >
      <h2>
        Click the Login Button to Log In!
      </h2>
    </div>
    <div
      className="build-title"
    />
    <div
      className="navbar"
    >
      <div
        className="button_things"
      >
        <Link
          to="/"
        >
          <i
            className="fas fa-home"
          />
           Home
        </Link>
        <Link
          to="/build"
        >
          <i
            className="fas fa-hammer"
          />
           Build!
        </Link>
      </div>
      <div
        className="mobile-button"
      >
        <Link
          to="/"
        >
          <i
            className="fas fa-home"
          />
           Home
        </Link>
        <Link
          to="/build"
        >
          <i
            className="fas fa-hammer"
          />
           Build!
        </Link>
        <div
          className="navigation-logged-out"
        >
          <Link
            to="/landing"
          >
            Log In 
            <i
              className="fas fa-sign-in-alt"
            />
          </Link>
        </div>
      </div>
      <div
        className="search-cont"
      >
        <lazy />
      </div>
      <div
        className="user_things"
      >
        <div
          className="navigation-logged-out"
        >
          <Link
            to="/landing"
          >
            Log In 
            <i
              className="fas fa-sign-in-alt"
            />
          </Link>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
